src/backend/gl/src/info.rs

```diff
@@ -55,28 +55,43 @@ impl Version {
     /// Note that this function is intentionally lenient in regards to parsing,
     /// and will try to recover at least the first two version numbers without
     /// resulting in an `Err`.
+    /// # Notes
+    /// `WebGL 2` version returned as `OpenGL ES 3.0`
     pub fn parse(mut src: &str) -> Result<Version, &str> {
-        // TODO: Parse version and optional vendor
         let webgl_sig = "WebGL ";
-        let is_webgl = src.contains(webgl_sig);
+        // According to the WebGL specification
+        // VERSION	WebGL<space>1.0<space><vendor-specific information>
+        // SHADING_LANGUAGE_VERSION	WebGL<space>GLSL<space>ES<space>1.0<space><vendor-specific information>
+        let is_webgl = src.starts_with(webgl_sig);
+        let is_es;
+
         if is_webgl {
-            return Ok(Version {
-                major: 2,
-                minor: 0,
-                is_embedded: true,
-                revision: None,
-                vendor_info: "".to_string(),
-            });
+            is_es = true;
+            let pos = src.rfind(webgl_sig).unwrap_or(0);
+            src = &src[pos + webgl_sig.len()..];
+        } else {
+            let es_sig = " ES ";
+            is_es = match src.rfind(es_sig) {
+                Some(pos) => {
+                    src = &src[pos + es_sig.len()..];
+                    true
+                }
+                None => false,
+            };
         }
 
-        let es_sig = " ES ";
-        let is_es = match src.rfind(es_sig) {
+        let glsl_es_sig = "GLSL ES ";
+        let is_glsl;
+        match src.find(glsl_es_sig) {
```

Comment:
same here, let's do it more functional style: `let is_glsl = if {... } else {..};`